lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1156,11 +1156,10 @@ SmallVector<unsigned> DotOperandEncodingAttr::getWarpsPerCTA() const {
 }
 SmallVector<unsigned> DotOperandEncodingAttr::getWarpOrder() const {
   // FIXME(Lezcano): Preexisting. Do we want to have this path at all?
-  if (mlir::isa<AMDMfmaEncodingAttr>(getParent())) {
+  if (mlir::isa<AMDMfmaEncodingAttr, AMDWmmaEncodingAttr, BlockedEncodingAttr>(
+          getParent())) {
     return ::getWarpOrder(getParent());
   }
-  // It's quite weird to talk about warp order when that the warps
-  // are broadcasted along the K dimension
```

Comment:
I've removed this comment, because warp order for dot operand actually affects how elements are stored.
let's say we have warp shape [2, 2], depending on order dot operand A will be shared between warp 0/1 and warp 2/3 or warp 0/2 and warp 1/3:

| :arrow_down:  M   :arrow_right: K | |
--- | ---
| warp 0/1 | warp 0/1 |
| warp 2/3 | warp 2/3 |

vs 

| :arrow_down:  M   :arrow_right: K | |
--- | ---
| warp 0/2 | warp 0/2 |
| warp 1/3 | warp 1/3 |

Reply:
I agree. It's just that you almost never want to write code so generic that would work for something that's broadcasted when using the order. In other words, it's something quite difficult to user right and quite easy to user wrong.

I would be happy to be convinced otherwise if you have an example tho.

Reply:
So, regarding the comment in l.1158. Do you think you really need the path above for AMD? Don't you think it'd be better to error out?